### ISSUE TITLE
RFE: report `export -f` as not portable

### ISSUE THREAD

**Issue by pabs3** [koalaman/shellcheck#1130]
#### For new checks and feature suggestions

- [?] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
  - I do not allow JavaScript
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet that shows the problem:

```sh

#!/bin/sh
foo () {
	echo foo
}
export -f foo
```

#### Here's what shellcheck currently says:

```
$ sh -n foo.sh
$ sh foo.sh
foo.sh: 5: export: Illegal option -f
$ shellcheck foo.sh
$ checkbashisms foo.sh
possible bashism in foo.sh line 5 (export only takes -p as an option):
export -f foo
$ bash -n foo.sh
$ bash foo.sh 
$ 
```

#### Here's what I wanted or expected to see:

Exporting functions is not portable to shells other than bash.

**Comment by koalaman** [comment]
Thanks for the suggestion, this has now been added!
